core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/TextInteractionDirective.js

```diff
@@ -0,0 +1,140 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for a multiple choice question in the simple editor.
+ */
+
+// NOTE TO DEVELOPERS: This is meant to be a reusable directive, so its only
+// dependencies should be standard utility services. It should not have any
+// concept of "state in an exploration".
+oppia.directive('textInteraction', [
+    'QuestionIdService', 'AnswerGroupObjectFactory', 'RuleObjectFactory',
+    'StatusObjectFactory', 'OutcomeObjectFactory', 'UrlInterpolationService',
+    function(QuestionIdService, AnswerGroupObjectFactory, RuleObjectFactory,
+          StatusObjectFactory, OutcomeObjectFactory, UrlInterpolationService) {
+      return {
+        restrict: 'E',
+        scope: {
+          // A unique ID that allows events to be broadcast specifically to this
+          // directive.
+          getUniqueId: '&uniqueId',
+          getCustomizationArgs: '&customizationArgs',
+          getAnswerGroups: '&answerGroups',
+          getRawDefaultOutcome: '&defaultOutcome',
+          saveCustomizationArgs: '&',
+          saveAnswerGroups: '&',
+          saveDefaultOutcome: '&',
+          addState: '&'
+        },
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/exploration_editor/simple_editor_tab/questions/' +
+          'text_interaction_directive.html'),
+        controller: [
+          '$scope', '$timeout',
+          function($scope, $timeout) {
+            // Note that a questionId generated in this way may contain spaces,
+            // since it is just the state name.
+            $scope.questionId = $scope.getUniqueId();
+            var answerGroups = $scope.getAnswerGroups();
+
+            $scope.getSubfieldId = function(label) {
+              return QuestionIdService.getSubfieldId($scope.questionId, label);
+            };
+
+            $scope.getFieldId = function(index) {
+              return $scope.questionId + '.' + index;
+            };
+
+            $scope.getAnswer = function() {
+              var newAnswerGroups = $scope.getAnswerGroups();
+              if (answerGroups.length !== 0) {
+                return newAnswerGroups[0].rules[0].inputs.x;
+              }
+            };
+
+            $scope.saveAnswer = function(newAnswer) {
+              var newAnswerGroups = answerGroups;
+              if(newAnswerGroups.length === 0 ) {
+                var newStateName = $scope.addState();
+                newAnswerGroups.push(AnswerGroupObjectFactory.createNew([
+                RuleObjectFactory.createNew('Contains', {
```

Comment:
Indentation of this and the next few lines is wrong.

Reply:
Done.